### PR TITLE
Migrate to new 1.8.0 rfdetr package

### LINKED ISSUE #6684: Resolve library dependency to rfdetr personal fork
The library has a dependency on a fork of `rfdetr` hosted in a personal GH repo ([source](https://github.com/open-edge-platform/training_extensions/blob/develop/library/pyproject.toml#L57)). To avoid depending on developers' personal github repos, it is recommend to do one of the following:

1.  Contribute the fixes to the original RF-DETR project and eventually switch to the official `rfdetr` package.
2. Use the original `rfdetr` package with a few monkey-patched methods to apply the necessary fixes

### PR DESCRIPTION
**Pull request by kprokofi**
### Summary

Migrate `library/` RF-DETR integration from a git-pinned fork (`bf69fe2d`) to the upstream `rfdetr==1.8.0` PyPI package.
### Changes
- `pyproject.toml`: pin `rfdetr==1.8.0`, 
- `common/rfdetr_mixin.py`: rebuild on `rfdetr.config` + `build_model_from_config` / `load_pretrain_weights`; download pretrained weights from `_pretrained_weights` URLs to `PRETRAINED_WEIGHTS_CACHE_DIR` (defaults to `~/.cache/torch/hub/checkpoints`); rank-0 download + `torch.distributed.barrier()` for DDP safety; use `load_pretrain_weights` (no PE interpolation / head resize / query trim duplication).
- `detection/rfdetr.py` & `instance_segmentation/rfdetr_inst.py`: per-variant `RFDETR*Config` mapping, Intel storage URLs.
- `detection/detectors/rfdetr.py`: background-class filtering in `postprocess` (model now outputs `num_classes + 1` logits).
- `detection/detectors/_rfdetr_vendored.py`: vendor `get_param_dict` + `compute_multi_scale_scales` from `rfdetr.training.param_groups` to avoid pulling `albumentations` (AGPL) and a second Lightning stack.
- `tests/unit/.../test_rfdetr.py`: update for new default variant and input sizes.
- `application/backend/.../manifests/detection/rfdetr_{s,m,l}.yaml` and `instance_segmentation/rfdetr_{s,m,l,xl}.yaml`: SHA sums verified against upstream files (Roboflow weights unchanged).

### Checklist

- [ ] The PR title and description are clear and descriptive
- [ ] I have manually tested the changes
- [ ] All changes are covered by automated tests
- [ ] All related issues are linked to this PR (if applicable)
- [ ] Documentation has been updated (if applicable)

### CODE REVIEW

File: library/src/getitune/backend/lightning/models/detection/rfdetr.py [hunk 1]
```diff
@@ -70,19 +75,17 @@ class RFDETR(RFDETRMixin, LightningDetectionModel):  # pyrefly: ignore[inconsist
     """
 
     _pretrained_weights: ClassVar[dict[str, str]] = {
-        "rfdetr_base": "https://storage.geti.intel.com/weights/rf-detr-base-2026.pth",
-        "rfdetr_large": "https://storage.geti.intel.com/weights/rf-detr-large-2026.pth",
-        "rfdetr_nano": "https://storage.geti.intel.com/weights/rf-detr-nano-2026.pth",
+        "rfdetr_nano": "https://storage.geti.intel.com/weights/rf-detr-nano.pth",
```
Comment (itallix) [inline library/src/getitune/backend/lightning/models/detection/rfdetr.py]: Why not `-2026` for nano?

**Review by itallix — DISMISSED** [review]

**Comment by kprokofi** [comment]
image

Predictions are correct as well

File: application/backend/app/execution/training/getitune_trainer.py [hunk 2]
```diff
@@ -425,15 +426,17 @@ def train_model(
...
             GetiTuneDeviceType.gpu if device.type is DeviceType.CUDA else GetiTuneDeviceType(device.type)
         )
+        class_path = model_cfg.get("class_path", "")
+        is_ultralytics = "ultralytics" in class_path
         engine_kwargs: dict[str, Any] = {
             "work_dir": self._data_dir / f"getitune-workspace-{model_id}",
             "device": getitune_device_type,
-            "checkpoint": weights_path,
         }
+        if is_ultralytics or has_model_revision:
```
Comment (itallix) [inline application/backend/app/execution/training/getitune_trainer.py]: Should it be just `has_model_revision` in condition? Don't we consider the case when ultralytics models do not have parent revision?

File: library/src/getitune/backend/lightning/models/common/rfdetr_mixin.py [hunk 3]
```diff
@@ -58,55 +62,52 @@ def _build_rfdetr_model(
...
+                os.environ.get(
+                    "PRETRAINED_WEIGHTS_CACHE_DIR",
+                    Path.home() / ".cache" / "torch" / "hub" / "checkpoints",
+                )
+            )
+            cache_dir.mkdir(parents=True, exist_ok=True)
+            local_path = cache_dir / Path(pretrain_url).name
+            is_distributed = torch.distributed.is_available() and torch.distributed.is_initialized()
+            rank = torch.distributed.get_rank() if is_distributed else 0
+            if rank == 0 and not local_path.exists():
```
Comment (itallix) [inline library/src/getitune/backend/lightning/models/common/rfdetr_mixin.py:112]: It looks like we don't need intermediate `rank` variable here and condition can still be built using `is_distributed` flag

```suggestion
            if not is_distributed and not local_path.exists():
```

File: library/src/getitune/backend/lightning/models/detection/detectors/_rfdetr_vendored.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2026 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Parts of this file are vendored from the rfdetr library (Apache-2.0):
+#   https://github.com/roboflow/rf-detr
+#
+# Vendored to avoid pulling rfdetr's ``train`` extra, which includes
+# AGPL-licensed ``albumentations`` and a second PyTorch Lightning stack.
```
Comment (itallix) [inline library/src/getitune/backend/lightning/models/detection/detectors/_rfdetr_vendored.py]: The comment regarding `albumentations` can probably be omitted. Their `train` extra specifies [`albumentations>=1.4.24,<3.0.0`](https://github.com/roboflow/rf-detr/blob/ba8cb2a21258470898b365d38b44ab23a8c1593b/pyproject.toml#L60), which is shipped under the MIT License - not `albumentationsx`, which is the one under the [AGPL](https://github.com/albumentations-team/albumentations?tab=readme-ov-file#your-options-moving-forward) license.

File: application/backend/app/execution/training/getitune_trainer.py (hunk 2 above)
Comment (kprokofi) [reply application/backend/app/execution/training/getitune_trainer.py]: No, it is correct. Ultralytics backend should always load weights in init, Lightning backend uses hardcoded weights and preloads them always.

File: library/src/getitune/backend/lightning/models/detection/detectors/_rfdetr_vendored.py (hunk 4 above)
Comment (kprokofi) [reply library/src/getitune/backend/lightning/models/detection/detectors/_rfdetr_vendored.py]: Yes, probably we can use upstream fully then

File: library/src/getitune/backend/lightning/models/common/rfdetr_mixin.py (hunk 3 above)
Comment (kprokofi) [reply library/src/getitune/backend/lightning/models/common/rfdetr_mixin.py:112]: This suggestion would break DDP — if we write if not is_distributed and not local_path.exists():, then in distributed mode rank 0 never downloads, and every rank blocks forever at the barrier.

File: application/backend/app/execution/training/getitune_trainer.py (hunk 2 above)
Comment (kprokofi) [reply application/backend/app/execution/training/getitune_trainer.py]: I added a comment regarding this

File: library/src/getitune/backend/lightning/models/common/rfdetr_mixin.py (hunk 3 above)
Comment (itallix) [reply library/src/getitune/backend/lightning/models/common/rfdetr_mixin.py:112]: Make sense, I have mistakenly thought it can't be 0 when `is_distributed = True`.

**Review by itallix — APPROVED** [review]
